public/scripts/project/metadata.js

```diff
@@ -127,17 +151,50 @@ define(function(require) {
       url: url
     });
     request.done(function(data) {
-      setMetadata(root, data, callback);
+      callback(data);
     });
     request.fail(function(jqXHR, status, err) {
       callback(err);
     });
   }
 
+  function loadAnonymousMetadata(config, callback) {
+    callback = lockSafeCallback(callback);
+
+    lock(function() {
+      fs.getxattr(root, PROJECT_META_KEY, function(err, value) {
+        if(err) {
+          if(err.code !== 'ENOATTR') {
+            return callback(err);
+          }
+          return setAnonymousMetadata(config, callback);
+        }
+
+        callback();
+      });
+    });
+  }
+
+  function loadMetadata(config, callback) {
+    if (config.user) {
+      return fetchMetadata(config, function(err, data) {
```

Comment:
This anonymous vs. regular metadata flow is too complicated.  Can you only have it break into two steps when you load/parse the data, but not when you write?  That is, `setMetadata()` should take the `metadata` to write, and do it in both cases vs. having two ways to write it.